sophia/src/term/_literal_kind.rs

```diff
@@ -1,54 +1,55 @@
 // this module is transparently re-exported by its parent `term`
 
-use std::hash::{Hash, Hasher};
-
 use super::*;
+use std::borrow::Cow;
+use std::hash::{Hash, Hasher};
 
 /// There are two kinds of literals: language-tagged, and typed.
 #[derive(Clone, Copy, Debug, Eq)]
-pub enum LiteralKind<T: AsRef<str>> {
+pub enum LiteralKind<T: TermData> {
     Lang(T),
-    Datatype(IriData<T>),
+    Datatype(Iri<T>),
 }
 pub use self::LiteralKind::*;
 
 impl<T> LiteralKind<T>
 where
-    T: AsRef<str>,
+    T: TermData,
 {
     /// Copy another literal kind with the given factory.
-    pub fn from_with<'a, U, F>(other: &'a LiteralKind<U>, mut factory: F) -> LiteralKind<T>
+    pub fn from_with<'a, U, F>(other: &'a LiteralKind<U>, mut factory: F) -> Self
```

Comment:
Shouldn't this be renamed as well?
```suggestion
    pub fn copy_with<'a, U, F>(other: &'a LiteralKind<U>, mut factory: F) -> Self
```

Reply:
I plan to do so but latter in the PR `Literal as own public type` :wink:.

BTW as suggested by you (https://github.com/pchampin/sophia_rs/pull/35#issuecomment-577721198) `LiteralKind` will then be removed anyway.